### ISSUE TITLE
src:prodserver fails with when using: --nolegacy_external_runfiles

### ISSUE THREAD

**Issue by jlisee** [alexeagle/angular-bazel-example#183]
Using bazel version 0.16.0 on commit e5560ae155c93f844112fc554603937d0a6a560b `bazel run --nolegacy_external_runfiles //src:prodserver` fails to launch.  The root issue appears to be issue in the nodejs rules and how they resolve paths, but I wanted to start in this project in case I missed something.

Here is a full reproduction:

```
$ yarn install
...
$ bazel run --nolegacy_external_runfiles //src:prodserver
INFO: Analysed target //src:prodserver (0 packages loaded).
INFO: Found 1 target...
Target //src:prodserver up-to-date:
  dist/bin/src/prodserver_bin.sh
  dist/bin/src/prodserver
INFO: Elapsed time: 0.450s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
/home/jlisee/.cache/bazel/_bazel_jlisee/a3cee343d71b0f2101c79d88606a48aa/execroot/angular_bazel_example/bazel-out/k8-fastbuild/bin/src/prodserver: line 139: source: filename argument required
source: usage: source filename [arguments]
```

From my digging it looks like the way paths our resolved in the nodejs rules it needs runtree layout like:

```
prodserver.runfiles/nodejs/bin/nodejs/bin/node
prodserver.runfiles/nodejs/bin/node
prodserver.runfiles/__main__/external/nodejs/bin/nodejs/bin/node
prodserver.runfiles/__main__/external/nodejs/bin/node
```

(**EDIT NOTE** fixed example path below)

Instead of just (which you get with the `nolegacy` flag) note the lack of `__main__/external`
```
prodserver.runfiles/nodejs/bin/nodejs/bin/node
prodserver.runfiles/nodejs/bin/node
```

**Comment by alexeagle** [comment]
Oh, thanks for the heads up. I didn't know there was a legacy layout in the runfiles, we'll need to adapt for it in rules_nodejs. Filed the linked issue there

**Comment by gregmagolan** [comment]
Thanks for the find. This should fix this issue: https://github.com/bazelbuild/rules_nodejs/pull/380

Also related is a fix for karma to work with this flag: https://github.com/bazelbuild/rules_typescript/pull/309/files

And a fix for protractor will be required as will in the angular/angular repo

**Comment by gregmagolan** [comment]
Resolved in #227
